jdaviz/tests/test_app.py

```diff
@@ -224,13 +224,51 @@ def test_to_unit(cubeviz_helper):
 
     cid = cubeviz_helper.app.data_collection[0].data.find_component_id('flux')
     data = cubeviz_helper.app.data_collection[-1].data
-    values = 1
+
+    # Surface brightness to flux
+
+    value = 1
     original_units = u.MJy / u.sr
     target_units = u.MJy
 
-    value = uc.to_unit(cubeviz_helper, data, cid, values, original_units, target_units)
+    value = uc.to_unit(cubeviz_helper, data, cid, value, original_units, target_units)
 
     assert np.allclose(value, 4.7945742429049767e-11)
 
+    # Flux to surface brightness
+
     original_units = u.MJy
     target_units = u.MJy / u.sr
+
+    value = uc.to_unit(cubeviz_helper, data, cid, value, original_units, target_units)
+
+    assert np.allclose(value, 1)
+
+    # Change from Fnu to Flam (with values shape matching spectral axis)
+
+    values = np.ones(3001)
+    original_units = u.MJy
+    target_units = u.erg / u.cm**2 / u.s / u.AA
+
+    new_values = uc.to_unit(cubeviz_helper, data, cid, values, original_units, target_units)
+
+    assert np.allclose(new_values,
+                       (values * original_units)
+                       .to_value(target_units,
+                                 equivalencies=u.spectral_density(cube.spectral_axis)))
+
+    # Change from Fnu to Flam (with a shape (2,) array of values indicating we
+    # are probably converting the limits)
+
+    values = [1, 2]
+    original_units = u.MJy
+    target_units = u.erg / u.cm**2 / u.s / u.AA
+
+    new_values = uc.to_unit(cubeviz_helper, data, cid, values, original_units, target_units)
+
+    # In this case we do a regular spectral density conversion, but using the
+    # first value in the spectral axis for the equivalency
+    assert np.allclose(new_values,

```

Comment:
Why not use `astropy.tests.helper.assert_quantity_allclose` here?

Reply:
new_values is a plain Numpy array